src/EventListener/AddFormatListener.php

```diff
@@ -58,7 +58,7 @@ public function onKernelRequest(GetResponseEvent $event)
         if (null === $routeFormat = $request->attributes->get('_format') ?: null) {
             $mimeTypes = array_keys($this->mimeTypes);
         } elseif (!isset($this->formats[$routeFormat])) {
-            throw new NotFoundHttpException('Not Found');
+            throw new NotFoundHttpException(sprintf('Format "%s" is not supported', $routeFormat));
```

Comment:
I'd say "415 Unsupported Media Type" or "406 Not Acceptable" but definitely not 404 👍

Reply:
https://httpstatuses.com/415 is about the payload of the request, so let's go with https://httpstatuses.com/406, which looks like a perfect fit for this

Reply:
@sroze in fact, 404 was right, and extensions shouldn't be used (in production, in development it's ok although I'm starting to think this causes more issues than it solves).